### DISCUSSION TITLE
Trigger

### DISCUSSION THREAD

**Discussion by trashhacker** [AvaloniaUI/Avalonia#17577] [Q&A]
Are u planing to implement triggers in avalonia

**Comment by stevemonaco** [comment]
There are UWP-style triggers as part of https://github.com/wieslawsoltes/Avalonia.Xaml.Behaviors/

Triggers are generally not the preferred approach in Avalonia. Style selectors eliminate the need for most. `DataTrigger` can be replaced by [binding to classes](https://docs.avaloniaui.net/docs/guides/data-binding/binding-classes) and then creating a class selector. The same approach can be used for styling a control based on a different (non-parent) control's state - class binding and then style the class. Though Avalonia is a bit awkward here.
